x-pack/filebeat/module/gcp/audit/config/pipeline.js

```diff
@@ -63,6 +63,35 @@ function Audit(keep_original_message) {
         fail_on_error: false,
     });
 
+    var setOrchestratorMetadata = function(evt) {
+          if (evt.Get("json.resource.type") === "k8s_cluster") {
+            evt.Put("orchestrator.type", "kubernetes");
+            // Dissect to extract the api_version
+            var dissect_processor = new processor.Dissect({
+              "tokenizer": "%{}/%{orchestrator.api_version}/%{}",
+              "field": "json.protoPayload.resourceName",
+              "target_prefix": "",
+            }).Run(evt);
+
+            var convert_processor = new processor.Convert({
+                fields: [
+                    {
+                        from: "json.resource.labels.cluster_name",
+                        to: "orchestrator.cluster.name",
+                        type: "string"
+                    },
+                    {
+                        from: "json.protoPayload.resourceName",
```

Comment:
As for this one, based on the example I don't think it should be the entire API path, it should just be the specific resource.  I would do something like below, which would probably replace the above dissect/grok.
```
- grok:
     field:
     patterns:
        - '/%{API_VERSION:orchestrator.api_version}/namespaces/%{DATA}/%{RESOURCE_TYPE:orchestrator.resource.type}(/%{HOSTNAME:orchestrator.resource.name})?'
        - '/%{API_VERSION:orchestrator.api_version}/%{RESOURCE_TYPE:orchestrator.resource.type}'
        - 'apis/%{RESOURCE_TYPE:orchestrator.resource.type}/%{API_VERSION:orchestrator.api_version}'
     pattern_definitions:
        API_VERSION: (v\d+([a-z]+)?(\d+)?)
        RESOURCE_TYPE: ([a-z]+((\.[a-z0-9]+)+)?)
```

Reply:
I took your comment into account and added some patterns to handle other example of logs.